src/infection-model/seir.tsx

```diff
@@ -307,31 +308,34 @@ export function getAllBracketCurves(inputs: CurveProjectionInputs) {
   ).forEach(([pop, cases, recovered, deaths], index) => {
     // distribute cases across compartments proportionally
 
-    const infectious = cases * pInitiallyInfectious;
+    const activeCases = cases - recovered - deaths;
+
+    const infectious = activeCases * pInitiallyInfectious;
     // exposed is related to the number of infectious
     // but it can't be more than the total uninfected population
     const exposed = Math.min(infectious * ratioExposedToInfected, pop - cases);
 
     singleDayState.set(index, seirIndex.susceptible, pop - cases - exposed);
     singleDayState.set(index, seirIndex.exposed, exposed);
     singleDayState.set(index, seirIndex.infectious, infectious);
-    singleDayState.set(index, seirIndex.mild, cases * pInitiallyMild);
-    singleDayState.set(index, seirIndex.severe, cases * pInitiallySevere);
+    singleDayState.set(index, seirIndex.mild, activeCases * pInitiallyMild);
+    singleDayState.set(index, seirIndex.severe, activeCases * pInitiallySevere);
     singleDayState.set(
       index,
       seirIndex.hospitalized,
-      cases * pInitiallyHospitalized,
+      activeCases * pInitiallyHospitalized,
     );
     singleDayState.set(
       index,
       seirIndex.mildRecovered,
-      cases * pInitiallyMildRecovered,
+      recovered * pInitiallyMildRecovered,
```

Comment:
If users haven't entered in any recovered cases yet the two recovered groups will be 0, which may be a weird UX when this change is initially deployed. Should I support both situations when users have zero recovered (use the current logic) and non-zero recovered (use the updated logic)?

Reply:
i agree, this does seem wrong ... I think maybe the condition to detect is when cases are defined but recovered is `undefined` ... that way if someone explicitly enters `0` for recovered we don't just ignore that value.

Reply:
I tried to add a check for "if recovered & deaths have been filled out vs if they are not filled out" to fall back to the old distribution logic if the user hasn't added recovered/death counts.